azure-client-runtime/src/main/java/com/microsoft/azure/v2/PollStrategy.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.v2;
+
+import com.microsoft.rest.v2.http.HttpRequest;
+import com.microsoft.rest.v2.http.HttpResponse;
+import rx.Single;
+
+import java.io.IOException;
+
+/**
+ * An abstract class for the different strategies that an OperationStatus can use when checking the
+ * status of a long running operation.
+ */
+abstract class PollStrategy {
+    private long delayInMilliseconds;
+
+    /**
+     * Get the number of milliseconds to delay before sending the next poll request.
+     * @return The number of milliseconds to delay.
+     */
+    final long delayInMilliseconds() {
+        return delayInMilliseconds;
+    }
+
+    /**
+     * Set the delay in milliseconds to 0.
+     */
+    final void clearDelayInMilliseconds() {
+        this.delayInMilliseconds = 0;
+    }
+
+    /**
+     * Update the delay in milliseconds from the provided HTTP poll response.
+     * @param httpPollResponse The HTTP poll response to update the delay in milliseconds from.
+     */
+    final void updateDelayInMillisecondsFrom(HttpResponse httpPollResponse) {
+        final String retryAfterSecondsString = httpPollResponse.headerValue("Retry-After");
+        if (retryAfterSecondsString != null && !retryAfterSecondsString.isEmpty()) {
+            try {
+                final long responseDelayInMilliseconds = Long.valueOf(retryAfterSecondsString) * 1000;
+                delayInMilliseconds = Math.max(responseDelayInMilliseconds, AzureProxy.defaultDelayInMilliseconds());
+            }
+            catch (NumberFormatException ignored) {
```

Comment:
In case parsing of `Retry-After` header fail should we fall back to `AzureProxy.defaultDelayInMilliseconds()`?

Reply:
I've changed this slightly so that PollStrategy gets passed an initial delayInMilliseconds value in its constructor. If Azure never sends a Retry-After header, then the PollStrategy will continue to use the initial delayInMilliseconds as its delay duration. If Azure ever sends a Retry-After header, then that will become the new delayInMilliseconds value and we'll forget about the previous delayInMilliseconds value. I think this is simpler and it should work fine.

Reply:
sounds good!